### PR TITLE
Remove inotifywait

### PR DESCRIPTION
**Pull request by rgrinberg**
We use inotify directly, there's no need for a command line backend.

### CODE REVIEW

**Comment by rgrinberg** [comment]
For now I repurposed the tests to use fswatch. It's not all that useful as we'll get rid of fswatch anyway, but it would be nice to keep the tests for our fsevents/inotify backends.

File: test/expect-tests/dune_file_watcher/dune_file_watcher_tests.ml [hunk 1]
```diff
@@ -106,16 +102,16 @@ let%expect_test _ =
...
+              | Fs_memo_event { path; kind = _ } -> path
+              | Watcher_terminated -> assert false)))
   in
   let print_events n = print_events ~try_to_get_events ~expected:n in
   Dune_file_watcher.wait_for_initial_watches_established_blocking watcher;
   Stdio.Out_channel.write_all "x" ~data:"x";
   print_events 1;
   [%expect {|
-x
+Timed out waiting for more events: expected 1, saw 0
```
Comment (jeremiedimino) [inline test/expect-tests/dune_file_watcher/dune_file_watcher_tests.ml:114]: What is going on here?

**Comment by jeremiedimino** [comment]
> For now I repurposed the tests to use fswatch. It's not all that useful as we'll get rid of fswatch anyway, but it would be nice to keep the tests for our fsevents/inotify backends.

Agreed. Just to make sure I understand what's happening:

- on Linux the test uses inotify
- on OSX it uses fswatch

Is that correct?

**Comment by jeremiedimino** [comment]
Why do we use fswatch in the Linux test rather than the inotify backend?

File: test/expect-tests/dune_file_watcher/dune_file_watcher_tests.ml (hunk 1 above)
Comment (rgrinberg) [reply test/expect-tests/dune_file_watcher/dune_file_watcher_tests.ml]: Fixed.

**Comment by rgrinberg** [comment]
> Why do we use fswatch in the Linux test rather than the inotify backend?

That was WIP. Now it's fixed. The Linux tests are now using inotify as expected.

For now, I disabled the macos tests. The fswatch based backend isn't working in the tests for now and it doesn't seem like  a good use of time to fix it's replacement is basically ready. This PR is therefore good to go.

**Comment by jeremiedimino** [comment]
Agreed

**Review by jeremiedimino — APPROVED** [review]
Looks good!

**Pull request merged**